contribute/code.rst

```diff
@@ -0,0 +1,211 @@
+
+You will have to learn how to use Git to edit the series of commits in your change and upload your new series. You will generally edit your commit series with an interactive git rebase. Then you will force-push your new series over the old one or make a new branch and open a new changeset. These are not perfect solutions, but they are the best we have.
+
+Luckily, there are graphical tools that make this job easier. Unluckily, if you don't understand what they are doing on the Git command line, you might mess up and have to start over from your old series. Even seasoned Git pros mess up this process sometimes, don't worry. There is almost always a way back to your old series.
+
+Each commit must continue to build, run, and pass tests after your changes.
+
+Format your commit message correctly
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Git gurus prefer that you keep the first line of your commit message (the summary) to 50 characters or less. They also prefer that no line in the commit message goes over 72 characters. It's fine if you have to break the rules — some changes can't be summarized in 50 characters and some links are longer than 72.
+
+Changeset description checklist
+-------------------------------
+
+Ensure that your changeset answers the following questions in its description.
+
+If you've followed the guidelines for commit messages, you can probably copy the relevant information from the commit messages into your changeset description. GitHub and GitLab will do this for you in single-commit changesets. A description that says "this is complicated to explain, please read the commit messages" is also fine. We trust you to strike the balance.
+
+What issue does your changeset resolve?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+All changes should be linked to an issue report that they resolve. It does not matter if you are adding a feature or fixing a problem. Is there an issue report filed on any repository which describes the problem you're fixing or the feature you're adding? Does the report specify which devices the issue occurs on or where the feature applies? Does it provide enough information that someone could look at it and know how to reproduce the problem or test the feature? If not, add this information to the issue report! Then, add a link to the issue report in your changeset description.
+
+For example:
+
+    Fixes https://github.com/ubports/ubuntu-touch/issues/1
+
+How did you test that the change was successful?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+All changes require testing to ensure they resolve the issue report they're referencing. On which devices did you test your change? How did you do that? Do you have any questions about existing functionality that you are unable to test? Make sure you include this in your changeset description. If you figured out how to reproduce an issue that has long been a mystery, add that to the issue report as well. It helps QA even if it means duplicate information.
+
```

Comment:
Same here:

All changes require testing to ensure they resolve the issue report they're referencing. Describe on which devices you did test your changes. Explain how you did did that and which channel (stable, release candidate, devel, edge) you where using. Name any questions about existing functionality that you where unable to test.